sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -144,6 +147,35 @@ case class Join(
     }
   }
 
+  def outerProjectList: Seq[NamedExpression] = {
```

Comment:
For select * from a natural join, we need to use a Project to get rid of redundant columns.

In mysql, select * from natural join would only contain one row if both size have the row of the same name. but user do can use something like `t1.a` or `t2.a` to explicitly reference the column here. Do you think we should implement it exactly the same? I'm afraid it would be a too complicated change, but I can do that if necessary.